WORKSPACE

```diff
@@ -32,6 +30,471 @@ load("@rules_haskell//tools:repositories.bzl", "rules_haskell_worker_dependencie
 # We don't use the worker mode, but this is required for bazel query to function.
 rules_haskell_worker_dependencies()
 
+load("//bazel_tools:os_info.bzl", "os_info")
+
+os_info(name = "os_info")
+
+load("@os_info//:os_info.bzl", "is_linux", "is_windows")
+
+http_archive(
+    name = "alex",
+    build_file_content = """
+load("@rules_haskell//haskell:cabal.bzl", "haskell_cabal_binary")
+haskell_cabal_binary(
+    name = "alex",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
+    """,
+    sha256 = "d58e4d708b14ff332a8a8edad4fa8989cb6a9f518a7c6834e96281ac5f8ff232",
+    strip_prefix = "alex-3.2.4",
+    urls = ["http://hackage.haskell.org/package/alex-3.2.4/alex-3.2.4.tar.gz"],
+)
+
+http_archive(
+    name = "happy",
+    build_file_content = """
+load("@rules_haskell//haskell:cabal.bzl", "haskell_cabal_binary")
+haskell_cabal_binary(
+    name = "happy",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
+    """,
+    sha256 = "9094d19ed0db980a34f1ffd58e64c7df9b4ecb3beed22fd9b9739044a8d45f77",
+    strip_prefix = "happy-1.19.11",
+    urls = ["http://hackage.haskell.org/package/happy-1.19.11/happy-1.19.11.tar.gz"],
+)
+
+http_archive(
```

Comment:
I guess there is no easier way to get a package from Hackage that isn’t in Stackage? This seems rather verbose and annoying to maintain.

Reply:
Actually I’m not sure why we can’t modify the `stack-snapshot.yaml` file for these?

Reply:
The reason is that this is an executable target, not a library target. That's what I meant by
> Binaries are not generated by stack_snapshot, so these are defined in WORKSPACE. Their library dependencies are supplied using the stack_snapshot.

I.e. the library `@stackage//:hpp` is generated by `stack_snapshot`, but the binary is defined separately.

Reply:
Ah I see makes sense, thanks for the explanation! That’s what I get for not reading the docs closely :slightly_smiling_face: